uberfire-project/uberfire-project-backend/src/main/java/org/guvnor/common/services/project/backend/server/WorkspaceProjectServiceImpl.java

```diff
@@ -280,6 +280,30 @@ public WorkspaceProject resolveProject(final Space space,
                               name);
     }
 
+    @Override
+    public WorkspaceProject resolveProject(final Space space,
+                                           final String projectName,
+                                           final String branchName) {
+
+
+        WorkspaceProject workspaceProject = resolveProject(space, projectName);
+
+        if (workspaceProject == null || branchName == null) {
+            return workspaceProject;
+        }
+
+        for (final Branch branch : workspaceProject.getRepository().getBranches()) {
+            if (branch.getName().equals(branchName)) {
+                return resolveProject(space, branch);
+            }
+        }
+
+        return new WorkspaceProject(workspaceProject.getOrganizationalUnit(),
```

Comment:
If the branch required wasn't found, it's incorrect to return the project anyway. Best to throw a exception here as well.

Reply:
done